### PR TITLE
AGW: mobilityD: DHCP: support to configure router IP via API

### LINKED ISSUE #3576: ip allocation: DHCP: add support for router ip and mac via API.

### PR DESCRIPTION
**Pull request by pshelar**
## Summary

There is a API to set router IP for static IP allocation. This
patch extends same API for UE IP allocated via DHCP.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/python/magma/mobilityd/ip_allocator_multi_apn.py [hunk 1]
```diff
@@ -26,19 +26,24 @@
...
+                 subscriberdb_rpc_stub,
+                 ip_allocator: IPAllocator):
         """ Initializes a Multi APN IP allocator
             This is wrapper around other configured Ip allocator. If subscriber
             has vlan configured in APN config, it would be used for allocating
             IP address by underlying IP allocator. For DHCP it means using
             vlan tag for DHCP request, for IP pool allocator it does not change
             behaviour.
         """
+        self.store = store
```
Comment (ardzoht) [inline lte/gateway/python/magma/mobilityd/ip_allocator_multi_apn.py]: `self._store = store`

File: lte/gateway/python/magma/mobilityd/subscriberdb_client.py [hunk 2]
```diff
@@ -77,31 +88,30 @@ def get_subscriber_ip(self, sid: str) -> Optional[StaticIPInfo]:
                 err.details())
         return None
 
-    def get_subscriber_apn_vlan(self, sid: str) -> int:
+    def get_subscriber_apn_networ_info(self, sid: str) -> NetworkInfo:
```
Comment (ardzoht) [inline lte/gateway/python/magma/mobilityd/subscriberdb_client.py]: typo in network

File: lte/gateway/python/magma/mobilityd/uplink_gw.py [hunk 3]
```diff
@@ -57,7 +64,7 @@ def read_default_gw(self):
                     default_gw[netifaces.AF_INET] is not None:
                 self.update_ip(default_gw[netifaces.AF_INET][0])
 
-    def update_ip(self, ip: Optional[str], vlan_id: Optional[str] = ""):
+    def update_ip(self, ip: Optional[str], vlan_id=None):
```
Comment (ardzoht) [inline lte/gateway/python/magma/mobilityd/uplink_gw.py:73]: we should add docs to this method and the ones below

File: lte/gateway/python/magma/mobilityd/subscriberdb_client.py [hunk 4]
```diff
@@ -20,30 +20,41 @@
...
+class NetworkInfo:
+    def __init__(self, gw_ip: Optional[str] = None, gw_mac: Optional[str] = None,
+                 vlan: int = 0):
+        gw_ip_parsed = None
+        try:
+            gw_ip_parsed = ipaddress.ip_address(gw_ip)
+        except ValueError:
+            logging.debug("invalid internet gw ip: %s", gw_ip)
+
+        self.gw_ip = gw_ip_parsed
```
Comment (ardzoht) [inline lte/gateway/python/magma/mobilityd/subscriberdb_client.py:32]: what if `gw_ip_parsed` can't be parsed and None is assigned? do we do additional checks for this?

**Review by ardzoht — CHANGES_REQUESTED** [review]

File: lte/gateway/python/magma/mobilityd/subscriberdb_client.py (hunk 4 above)
Comment (pshelar) [reply lte/gateway/python/magma/mobilityd/subscriberdb_client.py:32]: That is handled. in many cases the gw would be None.

**Review by ardzoht — APPROVED** [review]

**Pull request merged**